### PR TITLE
Yabasanshiro origin

### PR DESCRIPTION
**Pull request by devmiyax**
Setup script for YabaSanshiro SEGA Saturn emulator.
This is instruction video.
https://youtu.be/tgiJdss_sgQ

### CODE REVIEW

File: scriptmodules/emulators/yabasanshiro.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+rp_module_section="exp"
+rp_module_flags="!64bit"
+
+function depends_yabasanshiro() {
+    local depends=(cmake pkg-config python-pip cmake build-essential protobuf-compiler libprotobuf-dev libsecret-1-dev libssl-dev libsdl2-dev libboost-all-dev )
+    getDepends "${depends[@]}"
+}
+
+function sources_yabasanshiro() {
+    gitPullOrClone "$md_build" https://github.com/devmiyax/yabause.git pi4 bd9f235
```
Comment (dankcushions) [inline scriptmodules/emulators/yabasanshiro.sh:26]: I think it can just be `gitPullOrClone` without any params if you've set up rp_module_repo

**Review by dankcushions — COMMENTED** [review]
Thanks for this @devmiyax! Would be great to add it. I have some quick comments

File: scriptmodules/emulators/yabasanshiro.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="yabasanshiro"
+rp_module_desc="SEGA Saturn emulator Yaba Sanshiro"
+rp_module_help="ROM Extensions: .cue .chd\n\nCopy your SEGA Saturn ios images to $romdir/saturn"
+rp_module_licence="GPL2 https://github.com/devmiyax/yabause/blob/master/LICENSE"
+rp_module_repo="git https://github.com/devmiyax/yabause.git pi4"
+rp_module_section="exp"
+rp_module_flags="!64bit"
```
[the remark this answers is not in the document]
Comment (devmiyax) [reply scriptmodules/emulators/yabasanshiro.sh]: I want make this script just for pi4 or mc4.  
So change this line to ```rp_module_flags="rpi gles31"```

File: scriptmodules/emulators/yabasanshiro.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    gitPullOrClone "$md_build" https://github.com/devmiyax/yabause.git pi4 bd9f235
+    #git submodule update --init --recursive
+}
+
+function build_yabasanshiro() {
+    mkdir build
+    cd build
+    cmake ../yabause/ -DGIT_EXECUTABLE=/usr/bin/git -DYAB_PORTS=retro_arena -DYAB_WANT_DYNAREC_DEVMIYAX=ON -DYAB_WANT_ARM7=ON -DCMAKE_TOOLCHAIN_FILE=../yabause/src/retro_arena/pi4.cmake -DCMAKE_INSTALL_PREFIX="$md_inst"
+    make clean
+    make -j4
```
[the remark this answers is not in the document]
Comment (devmiyax) [reply scriptmodules/emulators/yabasanshiro.sh]: I removed -j4

File: scriptmodules/emulators/yabasanshiro.sh (hunk 1 above)
Comment (devmiyax) [reply scriptmodules/emulators/yabasanshiro.sh]: I removed params.

**Comment by devmiyax** [comment]
Thank you for your time @dankcushions .  I fixed.

**Comment by dankcushions** [comment]
Thank you!

Just out of interested, what's the purpose of the yabasanshiro folder and the 'backup.bin' that gets created in the `/RetroPie/roms/saturn` folder during install?

normally internal or BIOS related files or folders go into the system folder (`/RetroPie/BIOS` or `/RetroPie/BIOS/$system`), with only roms and save files in the /roms/$system folder. I am not sure what category this falls under. if it's an internal file but can't be redirected easily that's ok :)

I tried sega rally and it crashed when I selected arcade mode and as before it entered a race. Is this expected in this game? my log: https://pastebin.com/Q8zsMCH8

I also noticed it showed a mouse cursor in the top left corner - possible to hide this?

Thanks again!

**Comment by devmiyax** [comment]
>what's the purpose of the yabasanshiro folder and the 'backup.bin' that gets created in the /RetroPie/roms/saturn folder during install?

lr-yabasanshiro does create /RetroPie/roms/saturn/yabasanshiro/backup.bin. yabasanshiro does not.

**Comment by devmiyax** [comment]
>I also noticed it showed a mouse cursor in the top left corner - possible to hide this?
fixed in
https://github.com/devmiyax/yabause/commit/511fe48daff45c2cd497180803edbd94a6d07bab

**Comment by devmiyax** [comment]
>I tried sega rally and it crashed 

Sorry, I can't reproduce this issue. can you try the latest version?

**Comment by dankcushions** [comment]
> lr-yabasanshiro does create /RetroPie/roms/saturn/yabasanshiro/backup.bin. yabasanshiro does not.

yes, sorry you're right! no problem here, then.

mouse cursor is now gone, thanks!

i seem to have resolved the crash by getting the REDUMP version of sega rally - perhaps my previous copy was bad.

it seems to drop a few frames at points, and there's what looks like some depth issues where you can see bits of the sky through the ground, but I am using the older 19.3.2 MESA version, on Raspbian Buster, which I know has some issues, so this may be expected.

Looks good to me!

**Comment by dankcushions** [comment]
more testing:
- DecAthlete (USA) seems to either lock or crash after you select a player and it loads into the first 100m race.

Another thing, I notice the config files are kept in `/home/pi/.yabasanshiro` - is it possible to instead place them in `/opt/retropie/configs/saturn/yabasanshiro/` ? you can refer to this path in the buildscript via `"$md_conf_root/saturn/$md_id"`

if it's not possible to change the path, you instead could symlink so they're available in that path via something like this in `configure_yabasanshiro`:
```
mkUserDir "$md_conf_root/saturn"
mkUserDir "$md_conf_root/saturn/$md_id"
moveConfigDir "$home/.$md_id/" "$md_conf_root/saturn/$md_id"
```
I've not tested this! From memory...

**Comment by devmiyax** [comment]
Confirmed ~/.yabasanshiro is linked to /opt/retropie/configs/saturn/yabasanshiro with your script.

**Comment by devmiyax** [comment]
> DecAthlete (USA) seems to either lock or crash after you select a player and it loads into the first 100m race.

Yabasanshiro is not a perfect emulator. I'll improve it constantly.

**Comment by dankcushions** [comment]
Thank you, all looks good to me! Someone else will review it before final approval 👍

**Comment by joolswills** [comment]
Thanks.. I squashed the commits and merged it.

**Comment by joolswills** [comment]
Getting a build error when libopenal-dev is installed.

```
In file included from yabause/src/sndal.c:31:
yabause/src/threads.h:118:18: error: unknown type name ‘u64’
 int YabNanosleep(u64 ns);
```

Looks like it's missing an include for u64 typedef. I will force it to build without openal support.

**Comment by joolswills** [comment]
Applied the fixes mentioned

https://github.com/RetroPie/RetroPie-Setup/commit/8180ff56155f4b4278be8a5cbc12d811c4ca680a

**Pull request merged**